pkg/cmd/attestation/artifact/oci/oci.go

```diff
@@ -0,0 +1,110 @@
+package oci
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
+)
+
+var ErrDenied = errors.New("the provided token was denied access to the requested resource, please check the token's expiration and repository access")
+var ErrRegistryAuthz = errors.New("remote registry authorization failed, please authenticate with the registry and try again")
+
+type Client struct {
+	ParseReference func(string, ...name.Option) (name.Reference, error)
+	Get            func(name.Reference, ...remote.Option) (*remote.Descriptor, error)
+}
+
+func checkForUnauthorizedOrDeniedErr(err transport.Error) error {
+	for _, diagnostic := range err.Errors {
+		switch diagnostic.Code {
+		case transport.UnauthorizedErrorCode:
+			return ErrRegistryAuthz
+		case transport.DeniedErrorCode:
+			return ErrDenied
+		}
+	}
+	return nil
+}
+
+// where name is formed like ghcr.io/github/my-image-repo
+func (c Client) GetImageDigest(imgName string) (*v1.Hash, error) {
+	name, err := c.ParseReference(imgName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create image tag: %w", err)
+	}
+
+	desc, err := c.Get(name, remote.WithAuthFromKeychain(authn.DefaultKeychain))
```

Comment:
**Request for Change**

Can you write a comment here explaining the intent and behaviour of this keychain option. I think it ties back to the command docs that say:

> Note that if you provide an OCI URI for the artifact you must already be authenticated with a container registry.